api/v1/nodescan/service.go

```diff
@@ -273,17 +276,31 @@ func (s *serviceImpl) GetNodeVulnerabilities(_ context.Context, req *v1.GetNodeV
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	// Handle the new format of the request and scan node inventory additionally
+	// Scan Components containing pkgs from NodeInventory
 	if req.GetComponents() != nil {
-		layer, err := apiV1.GetVulnerabilitiesForComponents(s.db, req.GetComponents(), common.HasUncertifiedRHEL(req.GetNotes()))
-		if err != nil {
-			return nil, status.Error(codes.Internal, err.Error())
+		if resp.Features, err = s.getNodeInventoryVulns(req.GetComponents(), common.HasUncertifiedRHEL(req.GetNotes())); err != nil {
+			return nil, err
 		}
-		resp.Features = features.ConvertFeatures(layer.Features)
 	}
 	return resp, nil
 }
 
+func (s *serviceImpl) getNodeInventoryVulns(components *v1.Components, isUncertifiedRHEL bool) ([]*v1.Feature, error) {
+	log.Debugf("Scanning NodeInventory")
+	// Convert content sets to CPEs
+	cpes := s.repoToCPE.Get(components.GetRhelContentSets())
+	log.Debugf("Converted content sets '%v' to CPEs '%v'", components.GetRhelContentSets(), cpes)
+	for _, comp := range components.GetRhelComponents() {
+		comp.Cpes = append(comp.Cpes, cpes...)
+	}
+	layer, err := apiV1.GetVulnerabilitiesForComponents(s.db, components, isUncertifiedRHEL)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	log.Debugf("Scanned %d components from NodeInventory and returning %d features", len(components.GetRhelComponents()), len(layer.Features))
```

Comment:
Nit: I would adjust the terminology. We usually consider "Vulnerability Scanning" as a two parts process: Analysis (detection and inventory creation) and Vulnerability Matching. What you are doing here is the latter:

"Matched vulnerabilities on %d RHEL components in node inventory."

Also, would make this a INFO rather than DEBUG too much in the logs?  Maybe having this sort of info in the bundles would be valuable.

Reply:
Also, making this an INFO rather than a DEBUG would cause too much impact in the logs? Having this sort of info in the bundles would be valuable, but I am not sure of the impact. Just push me back if that is not feasible.